app/components/retriable_fast_image/index.tsx

```diff
@@ -3,22 +3,40 @@
 
 import React, {PureComponent} from 'react';
 import FastImage, {FastImageProps} from 'react-native-fast-image';
+import validator from 'validator';
 
 export const FAST_IMAGE_MAX_RETRIES = 3;
 
 type RetriableFastImageProps = FastImageProps & {
-    id: string
+    id?: string
 }
 
 type RetriableFastImageState = {
-    retry: number
+    retry: number,
 }
 
 export default class RetriableFastImage extends PureComponent<RetriableFastImageProps, RetriableFastImageState> {
     state = {
         retry: 0,
     }
 
+    componentDidMount() {
+        if (this.props.source && typeof this.props.source != 'number' && this.props.source.uri) {
+            const config: validator.IsURLOptions = {
+                protocols: ['file', 'http', 'https'],
+                require_host: false,
+                require_tld: false,
+            };
+
+            // If not a valid error, show error
+            if (!validator.isURL(this.props.source.uri, config)) {
```

Comment:
I would add a prop to render a broken image if the prop is set, otherwise on not valid it should render null. this here is not preventing the app from crashing.

Also the validation should include a DataURI

Reply:
about rendering the broken image prop, components that implement this retriable_fast_image component should specify it if needed.

Reply:
I've added `renderOnError` prop - what should the image be?